Stacey Brosnan/week01/Objects/js/exercises.js

```diff
@@ -0,0 +1,219 @@
+    }
+    };
+
+      const areaTriangle = function ( triangle ) {
+        const a = triangle.sideA;
+        const b = triangle.sideB;
+        const c = triangle.sideC;
+
+        const p = ( a + b + c ) / 2 ;
+        let area = Math.sqrt(p * ( p - a ) * ( p - b ) * (p - c ));
+        return area;
+  };
+
+    const isObtuse = function ( triangle ) {
+      const aSquared = triangle.sideA * triangle.sideA;
+      const bSquared = triangle.sideB * triangle.sideB;
+      const cSquared = triangle.sideC * triangle.sideB;
+
+      if ( cSquared === aSquared + bSquared ) {
+        return `This triangle is a right triangle. It is not obtuse.`;
+      } else if ( cSquared < aSquared + bSquared ) {
+        return `This triangle is an acute triangle. It is not obtuse.`;
+      } else if ( cSquared > aSquared + bSquared ) {
+        return `This triangle is an obtuse triangle`;
+      }
+    };
+
+    console.log( `${isEquilateral(triangleA)}` );
+    console.log( `${isIsosceles(triangleA)}` );
+    console.log( `${areaTriangle(triangleA)}` );
+    console.log( `${isObtuse(triangleA)}` );
+
+    // # The Cash Register
+    //
+    // Write a function called cashRegister that takes a shopping cart object. The object contains item names and prices (itemName: itemPrice). The function should return the total price of the shopping cart.
+
+    const cartForParty = {
+      banana: 1.25,
+      handkerchief: 0.99,
+      Tshirt: 25.01,
+      apple: 0.60,
+      nalgene: 10.34,
+      proteinShake: 22.36
+   };
+
+    const cashRegister = function ( shopping ) {
+      const prices = Object.values( shopping );
+      let sum = 0;
+      for (let i = 0; i < prices.length; i++ ) {
+      cost = prices[i];
+      sum = sum + cost;
+      }
+      return sum;
+    };
+
+    console.log( `${cashRegister(cartForParty)}` );
+
+//   # Credit Card Validation
+//
+// You're starting your own credit card business. You've come up with a new way to validate credit cards with a simple function called validateCreditCard that returns true or false.
+//
+// Here are the rules for a valid number:
+//
+// - Number must be 16 digits, all of them must be numbers
+// - You must have at least two different digits represented (all of the digits cannot be the same)
+// - The final digit must be even
+// - The sum of all the digits must be greater than 16
+
+
+// const validateCreditCard = function (num) {
+//   const price = num.charAt(10)
+//   console.log(price);
+//   };
+//   console.log(validateCreditCard(`9999-9999-8888-0000`));
+
+
+    const validateCreditCard = function (num) {
+            let onlyNum = num.replace(/\D/g, '');//remove all non digit characters
+            // num = num.replace(/\-/, ''); - not using the wildcard correctly?
+
+            if (onlyNum.length === 16) { //number must have 16 digits
+                let sum = 0;
+                for (let i =0; i < onlyNum.length; i++) {
+                    let digit = parseInt(onlyNum.charAt(i));
```

Comment:
.length doesn't work on a number. 
A quick little tip around this is you could do `onlyNum.toString().length` This will turn onlyNum into a string and then look at the length of the string. This won't change the variable onlyNum into a string, but for that line of code it will treat onlyNum as a string (don't worry about this concept too much at the moment, it will make more sense later, but thought it was worth mentioning in this instance).